Feature request: markedDays props

bluenex opened the pull request:
Hi,

I'm looking for a React calendar library that is minimal and simple. This project looks very well suit to what I need. However, it lacks features I would like to use so here's my PR. I think it could be useful to have `markedDays` props so that we can mark days with events or something like that on the calendar. Here is an image of my implementation:

![image](https://user-images.githubusercontent.com/11027706/60314607-5fe82a80-998d-11e9-90e4-753e27374e26.png)

However, I think this will need some more tweaks:

- ~white mark color on today is looking better~
- ~how should we implement different mark colors on today and other days?~

plus:

- do you guys think it's good to add `onClick` prop? I'm thinking about clicking on the day and it returns the day being click, something like:

```js
onClick: day => doSomething(day)
```

This way when the user click we can hook the navigation function on it to the certain day.

Thank you for a very neat project again! 😄

bluenex committed:
add markedDays props

bluenex committed:
use opposite colors on markedDay of today and other days

arncet commented:
Hi @bluenex, thank you for your feedback and your contribution.
`markedDays` is a good idea of feature.
I putted some comments on your PR.
Callback methods like `onClick` (or `onChangeMonth` ...) can be really helpful, i'll try to add those soon

arncet commented on src/components/Calendar.js:
```diff
@@ -101,7 +101,7 @@ class Calendar extends PureComponent {
           {getDays(firstDayToDisplay).map(day =>
             <div
-              className={`rlc-day ${this.getClassNames(day)}`}
+              className={`rlc-day ${this.getClassNames(day)}${markedDays && markedDays.map(x => getDateWithoutTime(x)).includes(day) ? ' marked-day' : ''}`}
```
All classnames are generated by `getClassNames` method, so you can manage `marked-day` on this method

arncet commented on src/components/Calendar.js:
```diff
@@ -133,7 +134,8 @@ Calendar.propTypes = {
   monthLabels: arrayOf(string),
-  timezone: string
+  timezone: string,
+  markedDays: arrayOf(number)
```
Don't forget to update `README.md`

arncet replied:
I think that `markedDays` can also be a function (`oneOfType([arrayOf(number), func])`), like `disableDates` for more flexibility.

bluenex committed:
refactor(Calendar.js): handle marked day classname in getClassNames

also change name following the convention

bluenex committed:
refactor(Calendar.js): allow markedDays as function

bluenex committed:
docs(README.md): include markedDays props in README

bluenex committed:
feat(Calendar.js): add onClickDate prop

bluenex commented:
@arncet 

Hi,

Sorry for drag it for long, I almost forget about this PR. I've fixed all according to your suggestion. In the latest commit (7ee10f5), I added `onClickDate` prop which works fine to me but I'm not sure if it is ok for you. Could you please check it out?

Sorry again that this has been too long, I hope we can merge this soon!

arncet commented:
Hi @bluenex, thank you for your feedback, your changes will soon be merged.
